internal/oracles/mgr.go

```diff
@@ -50,18 +50,25 @@ func NewOracleMgr(config map[string]map[string]string, eventStore *events.EventS
 
 // Start starts the oracle manager.
 // It will block until Stop is called.
-func (omgr *OracleMgr) Start() {
+func (omgr *OracleMgr) Start() error {
 	ctx, cancel := context.WithCancel(context.Background()) // context that will be canceled when the manager shuts down
 	omgr.cancel = cancel
 	// Listen for status changes
 	// Start oracles if the node is a validator and is caught up
 	// Stop the oracles, if the node is not a validator
-	go func() {
-		// cancel function for the oracle instance
-		// if it is nil, then the oracle is not running
-		var oracleInstanceCancel context.CancelFunc
-
-		for {
+	// cancel function for the oracle instance
+	// if it is nil, then the oracle is not running
+	var oracleInstanceCancel context.CancelFunc
+
+	var errChan = make(chan error, 1)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case err := <-errChan:
+			return err
+		default:
 			// still in the catch up mode, do nothing
 			if omgr.cometNode.IsCatchup() {
```

Comment:
Not new, but I have recently noticed intense cpu use on startup until sync is achieved, which can be forever until a peer is established.  This loop has no delays, it just calls IsCatchup as fast as possible.

Reply:
Oops, this is supposed to have a 1s delay in the default case

Reply:
It does have a 1s delay, at the end of the default case

Reply:
Yeah, that is too late though because it does `continue` in this `IsCatchup` branch.  I would suggest something along the lines of:


```go
	for {
		select {
		case <-ctx.Done():
			return nil
		case err := <-errChan:
			return err
		case <-time.After(time.Second):
		}
		
		// still in the catch up mode, do nothing
		if omgr.cometNode.IsCatchup() {
			continue
		}
...
```

Reply:
Just fixed